### PR TITLE
feat(onto creation): allow that resources and properties are not sorted by inheritance (DEV-626)

### PR DESCRIPTION
**Pull request by jnussbaum**
resolves DEV-626

### CODE REVIEW

**Comment by irinaschubert** [comment]
Should the "not" in the title of the PR be removed?

**Comment by jnussbaum** [comment]
No, it's correct. During onto creation, the res & props need not be sorted by inheritance. Do you think there is a better title with less misunderstandings?

File: knora/dsplib/utils/onto_create_ontology.py [hunk 1]
```diff
@@ -298,6 +299,68 @@ def create_users(con: Connection, users: list[dict[str, str]], groups: dict[str,
...
+
+    Returns:
+        the sorted list of resource classes, in the format as it was in the JSON file
+    """
+
+    onto_name = ontology['name']
+    unsorted_resources: list[dict[Any, Any]] = ontology['resources']
+    sorted_resources: list[dict[Any, Any]] = list()
+    ok_resource_names: list[str] = list()
+    while len(unsorted_resources) > 0:
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply knora/dsplib/utils/onto_create_ontology.py:318]: Yes. The for-loop needs to repeat until there are no more unsorted resources. One for-loop alone cannot do it.

File: knora/dsplib/utils/onto_create_ontology.py [hunk 2]
```diff
@@ -298,6 +299,68 @@ def create_users(con: Connection, users: list[dict[str, str]], groups: dict[str,
...
+            res_name = f'{onto_name}:{res["name"]}'
+            parent_classes = res['super']
+            if isinstance(parent_classes, str):
+                parent_classes = [parent_classes]
+            parent_classes = [re.sub(r'^:([^:]+)$', f'{onto_name}:\\1', elem) for elem in parent_classes]
+            parent_classes_okay = [not parent.startswith(onto_name) or parent in ok_resource_names for parent in parent_classes]
+            if all(parent_classes_okay):
+                sorted_resources.append(res)
+                ok_resource_names.append(res_name)
+                unsorted_resources.remove(res)
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply knora/dsplib/utils/onto_create_ontology.py:329]: If only one parent class is not okay, I cannot declare a resource as okay. In such a case, the for-loop will continue, and in one of the later passes, the not-okay parent will become okay.

**Comment by jnussbaum** [comment]
Hi Irina, thanks for your comments. I responded to them and added unit tests. Can you re-review it?

**Review by irinaschubert — APPROVED** [review]

**Pull request merged**